src/storage/write_ahead_log/log_serializer_task.cpp

```diff
@@ -148,26 +184,44 @@ void LogSerializerTask::HandFilledBufferToWriter() {
   NOISEPAGE_ASSERT(primary_replication_manager_ != DISABLED ||
                        retention_policy != transaction::RetentionPolicy::RETENTION_LOCAL_DISK_AND_NETWORK_REPLICAS,
                    "If replication is disabled, then you can't send buffers to replicas.");
+  if (destination == SerializeDestination::DISK) {
+    if (disk_filled_buffer_ != nullptr) {
```

Comment:
This is probably echoing @mbutrovich comment's a bit, but it's unclear to me why we'd call `HandFilledBufferToWriter` with a destination of `DISK` if the disk filled buffer was null (i.e. not filled). If this is a legitimate scenario I think a comment would go a long way.